### DISCUSSION TITLE
Authentication required / 403 No valid crumb ...

### DISCUSSION THREAD

**Discussion by ololologist** [jenkinsci/bitbucket-push-and-pull-request-plugin#210] [Q&A]
- Jenkins **2.289.1**
- Bitbucket Push and Pull Request Plugin **2.7.2**
- smee.io

Hello, 
we are trying to trigger our SeedJob when a PR is merged to the master branch in our  BitBucket repo. Here is what has been done. 

### Jenkins 
Our Jenkins is running inside a private subnet in AWS VPC on Ubuntu Server 20.04.2 LTS, so we have tried to follow [these instruction](https://www.jenkins.io/blog/2019/01/07/webhook-firewalls/) and apply smee.io delivery service. 
We have installed smee-client and we use the following line to bind it with Jenkins webhook endpoint
`# smee --url https://smee.io/%smee_string% --target http://127.0.0.1:8080/bitbucket-hook/`
`Forwarding https://smee.io/%smee_string% to http://127.0.0.1:8080/bitbucket-hook/`
`Connected https://smee.io/%smee_string%`

### BitBucket 
We have created a webhook in the BB repo
BitBucket -> Repository -> Repository Settings -> Webhooks -> Add webhook  
<img width="1003" alt="Screen Shot 2021-09-01 at 5 44 08 PM" src="https://user-images.githubusercontent.com/39486730/131692369-864be706-d22c-47d8-aff2-1557a75325d8.png">

and we can see that the payloads can actually reach smee.io
![Screen Shot 2021-09-01 at 5 47 21 PM](https://user-images.githubusercontent.com/39486730/131692949-2cf4c57e-aac9-4adc-936d-c218adae8363.png)
 
Anyway when a BB event payload I can see following errors in the log
` Error: cannot POST /bitbucket-hook/ (403)`
`...`
`Error 403 No valid crumb was included in the request`
[full log](https://pastebin.com/J3qcvcF4) 

After adding ` -Dhudson.security.csrf.GlobalCrumbIssuerConfiguration.DISABLE_CSRF_PROTECTION=true` to `/etc/default/jenkins` and restarting Jenkins the error log shows the following 

`Error: cannot POST /bitbucket-hook/ (403)`
`...`
`Authentication required`
[full log ](https://pastebin.com/NFhEEf6e)

Could anyone please kindly guide me to any solution possible?

**Comment by derhecht** [comment]
read https://confluence.atlassian.com/bitbucketserver/manage-webhooks-938025878.html already? what is the output of your payload check in BB?
on which server/service is the 403 produced?

**Comment by Srinirap** [comment]
I am also hitting the same issue, Could you please solve this. I guess the issue is the URL is not working, I have created the hook http://<my jenkins>/myhook/ when I try to enter the url on the browsert it's reporting not found

`
URI: | /myhook/
-- | --
404
Not Found
Stapler

<br class="Apple-interchange-newline">`
